src/components/map/MapSearch.js

```diff
@@ -21,16 +31,26 @@ function callback(key) {
 }
 
 const MapSearch = () => {
+  const mapRef = useRef();
+  
+  const handleViewportChange = useCallback(
+    newViewport => setViewport(newViewport),
+    []
+  );
   const [filterDataList, setFilterDataList] = useState([]);
   const [searchText, setSearchText] = useState('');
-  const [long, setLong] = useState();
-  const [lat, setLat] = useState();
-  const [viewport, setViewport] = useState();
-  const [activeFilters, setActiveFilters] = useState ();
+  const [long, setLong] = useContext(ContextLong);;
```

Comment:
I still think we should be using a state wrapper that isn't dependent upon an additional dependency but once we do that we should look at these states and see which of them change together and therefore should just be on one object together.